lib/active_fedora/core_core.rb

```diff
@@ -0,0 +1,67 @@
+module ActiveFedora
+  module CoreCore
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def initialize_generated_modules # :nodoc:
+        generated_association_methods
+      end
+
+      def generated_association_methods
+        @generated_association_methods ||= begin
+          mod = const_set(:GeneratedAssociationMethods, Module.new)
+          include mod
+          mod
+        end
+      end
+    end
+
+    def ldp_source
+      @ldp_source
+    end
+
+    # Returns true if +comparison_object+ is the same exact object, or +comparison_object+
+    # is of the same type and +self+ has an ID and it is equal to +comparison_object.id+.
+    #
+    # Note that new records are different from any other record by definition, unless the
+    # other record is the receiver itself.
+    #
+    # Note also that destroying a record preserves its ID in the model instance, so deleted
+    # models are still comparable.
+    def ==(other)
```

Comment:
Note that the definition of `==` is slightly different than before, but now matches upstream.